R/utils-numeric.R

```diff
@@ -19,4 +19,82 @@ nonZeroRound <- function(x, digits) {
   } else { 
     round(x,digits) 
   } 
+}
+
+#' Replace numeric NAs with 0 - update by reference
+#'
+#' This function replaces NAs in numeric columns with 0.
+#' @param x data.table, data.frame, or list
+#' @param cols vector of column names for which the NA replacement should occur.
+#' Default is all numeric columns.
+#' @return x with desired NAs replaced with 0.
+#' @importFrom purrr map_lgl
+#' @export
+setNaToZero <- function(df, cols = NULL) {
+
+
+  if (is.null(cols)) {
+    # if cols not set, use all numeric cols
+    cols <- names(df)[purrr::map_lgl(df, is.numeric)]
+  } else {
+    # Validate that all specified columns are numeric
+    if (!all(purrr::map_lgl(df[cols], is.numeric))) stop('All columns must be numeric')
+  }
+
+  data.table::setnafill(df, fill = 0, cols = cols)
+}
+
+
+#' Replace numeric NAs with 0
+#'
+#' This function replaces numeric NAs with 0.
+#' @param x list, data.frame, array, or vector
+#' @param cols Optional. When appropriate, vector of column names 
+#' for which the NA replacement should occur.
+#' Default is all numeric columns.
+#' @return x with desired NAs replaced with 0.
+#' @export
+naToZero <- function(x, ...) {
+  UseMethod("naToZero")
+}
+
+#' @importFrom purrr map_lgl
+#' @export
+naToZero.list <- function(x, cols = NULL) {
+
+  # if cols not set, use all numeric cols
+  if (is.null(cols)) {
```

Comment:
you might write an internal helper fxn for this.. something like 'findAndValidateNumericCols', rather than have the duplicate code here.